### PR TITLE
refactor: make type-specific data (such as StructInfo) part of TypeInfo

### PR DESCRIPTION
**Pull request by Wodann**
Required for https://github.com/mun-lang/mun/pull/287.

### CODE REVIEW

**Review by baszalmstra — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,9 @@
 cmake_minimum_required(VERSION 3.10)
 project(MunRuntime VERSION 0.1.0 LANGUAGES C CXX)
 
+option(mun_build_examples "Build all of Mun's own examples." OFF)
```
Comment (baszalmstra) [inline CMakeLists.txt:4]: Why the change of capitalization? I always use capital variable names. I think thats common practice but cant find the source.

**Review by baszalmstra — CHANGES_REQUESTED** [review]
Looks good except for the one question!

File: CMakeLists.txt (hunk 1 above)
Comment (Wodann) [reply CMakeLists.txt:4]: I saw that we used a mix of capitalised and non-capitalised and while looking at `googletest` noticed that they use lowercase too, so went with that.

For reference, https://github.com/google/googletest/blob/master/googletest/CMakeLists.txt

Comment (Wodann) [reply CMakeLists.txt:4]: Test webhook
